src/main/java/com/ericsson/eiffel/remrem/publish/service/MessageServiceRMQImpl.java

```diff
@@ -1,101 +1,138 @@
+        } catch (final JsonSyntaxException e) {
             String resultMsg = "Could not parse JSON.";
             if (e.getCause() != null) {
                 resultMsg = resultMsg + " Cause: " + e.getCause().getMessage();
             }
             log.error(resultMsg, e.getMessage());
-            List<SendResult> results = new ArrayList<>();
-            results.add(new SendResult(resultMsg));
-            return results;
-    	}
+            List<PublishResultItem> events = new ArrayList<>();
+            createFailureResult(events);
+            return new SendResult(events);
+        }
     }
     
-    /* (non-Javadoc)
-     * @see com.ericsson.eiffel.remrem.publish.service.MessageService#send(java.lang.String, com.google.gson.JsonElement)
+    /*
+     * (non-Javadoc)
+     * @see com.ericsson.eiffel.remrem.publish.service.MessageService#send(com.google.gson.JsonElement, com.ericsson.eiffel.remrem.protocol.MsgService, java.lang.String)
      */
-    @Override public List<SendResult> send(String routingKey, JsonElement json) {
-    	List<String> msgs = new ArrayList<>();
-    	
-    	if (json == null) {    		
-            String resultMsg = "Invalid json content.";
-            log.error(resultMsg);
-            List<SendResult> results = new ArrayList<>();
-            results.add(new SendResult(resultMsg));
-            return results;
-    	}
-
-    	if (json.isJsonArray()) {
-    		JsonArray bodyJson = json.getAsJsonArray();
+    @Override
+    public SendResult send(JsonElement json, MsgService msgService, String userDomain) {
+        
+        Map<String, String> map = new HashMap<>();
+        Map<String, String> routeKeyMap = new HashMap<>();
+        List<PublishResultItem> events = new ArrayList<>();
+        if (json == null) {
+            createFailureResult(events);
+        }
+        if (json.isJsonArray()) {
+            JsonArray bodyJson = json.getAsJsonArray();
+            for (JsonElement obj : bodyJson) {
+                getAndCheckEvent(msgService, map, events, obj,routeKeyMap,userDomain);
+            }
+        } else {
+            getAndCheckEvent(msgService, map, events, json,routeKeyMap,userDomain);
+        }
 
-    		for (JsonElement obj : bodyJson) {
-    			msgs.add(obj.toString());
-    		}
-    	} else {
-    		msgs.add(json.toString());
-    	}
-    	
-        return send(routingKey, msgs);
+        if (map.size() > 0) {
+            SendResult result = send(routeKeyMap, map);
+            events.addAll(result.getEvents());
+            result.setEvents(events);
+            return result;
+        } else {
+            SendResult result = new SendResult();
+            result.setEvents(events);
+            return result;
+        }
     }
     
-    private SendResult sendMessage(String routingKey, String msg) {
-        String resultMsg = SUCCEED;
+    private String sendMessage(String routingKey, String msg) {
+        String resultMsg = PropertiesConfig.SUCCEED;
```

Comment:
I have always wondered why we use "succeed" instead of "success". Feels like the wrong term to use.